### ISSUE TITLE
Data Group record modifications fail 

### ISSUE THREAD

**Issue by LeonardosGitHub** [F5Networks/f5-appsvcs-extension#378]
### Environment
 * Application Services Version: Tested 3.16 & 3.23
 * BIG-IP Version: 15.1.0.4

### Summary
When managing a Data Group with AS3:
Once the Data Group is created and one of the records contain a colon (:).  When we attempt to modify the Data Group it fails with an error "Cannot read property '0' of null"

### Steps To Reproduce
1. Submit the following declaration to create Data Group:
```json
{
  "Common": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "datagroup": {
        "class": "Data_Group",
        "keyDataType": "string",
        "records": [
          {
            "key": "f5.com:443"
          },
          {
            "key": "f5.com"
          }
        ]
      }
    }
  },
  "class": "ADC",
  "schemaVersion": "3.12.0",
  "label": "ltm",
  "id": "ltm2",
  "remark": "ltm3",
  "updateMode": "selective",
  "controls": {
    "archiveTimestamp": "2020-01-29T01:27:54.691Z",
    "trace": true,
    "logLevel": "debug",
    "traceResponse": true
  }
}
```
2. Submit the following declaration to update the Data Group:
```json
{
  "Common": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "datagroup": {
        "class": "Data_Group",
        "keyDataType": "string",
        "records": [
          {
            "key": "example.com"
          },
          {
            "key": "f5.com"
          }
        ]
      }
    }
  },
  "class": "ADC",
  "schemaVersion": "3.12.0",
  "label": "ltm",
  "id": "ltm2",
  "remark": "ltm3",
  "updateMode": "selective",
  "controls": {
    "archiveTimestamp": "2020-01-29T01:27:54.691Z",
    "trace": true,
    "logLevel": "debug",
    "traceResponse": true
  }
}
```
3. Observe the following error response:
```json
{
  "results": [
    {
      "message": "Cannot read property '0' of null",
      "host": "localhost",
      "tenant": "Common",
      "code": 422
    }
  ]
}
```

### Expected Behavior
Expect the Data Group to be updated to the new values.

### Actual Behavior
An error occurs and the Data Group remains at the original declaration value.

```json
{
  "results": [
    {
      "message": "Cannot read property '0' of null",
      "host": "localhost",
      "tenant": "Common",
      "code": 422
    },
    {
      "message": "Cannot read property '0' of null",
      "host": "localhost",
      "tenant": "Common",
      "code": 422
    }
  ],
  "declaration": {
    "Common": {
      "class": "Tenant",
      "Shared": {
        "class": "Application",
        "template": "shared",
        "datagroup": {
          "class": "Data_Group",
          "keyDataType": "string",
          "records": [
            {
              "key": "example.com"
            },
            {
              "key": "f5.com"
            }
          ]
        }
      }
    },
    "class": "ADC",
    "schemaVersion": "3.12.0",
    "label": "ltm",
    "id": "ltm2",
    "remark": "ltm3",
    "updateMode": "selective",
    "controls": {
      "archiveTimestamp": "2020-11-11T22:29:34.192Z",
      "trace": true,
      "logLevel": "debug",
      "traceResponse": true
    }
  },
  "code": 422
}

**Comment by dstokesf5** [comment]
This is in our internal product backlog as AUTOTOOL-2057.

**Comment by dstokesf5** [comment]
This issue has been resolved as part of the 3.25.0 release.
